lib/msal-common/src/authority/Authority.ts

```diff
@@ -220,13 +234,24 @@ export class Authority {
      */
     public get jwksUri(): string {
         if(this.discoveryComplete()) {
-            const endpoint = this.replacePath(this.metadata.jwks_uri);
-            return this.replaceTenant(endpoint);
+            return this.replacePath(this.metadata.jwks_uri);
         } else {
             throw ClientAuthError.createEndpointDiscoveryIncompleteError("Discovery incomplete.");
         }
     }
 
+    /**
+     * Returns a flag indicating that tenant name can be replaced in authority {@link IUri}
+     * @param authorityUri {@link IUri}
+     * @private
+     */
+    private canReplaceTenant(authorityUri: IUri): boolean {
+        return authorityUri.PathSegments.length === 1
+            && !Authority.reservedTenantDomains.has(authorityUri.PathSegments[0])
+            && this.getAuthorityType(authorityUri) === AuthorityType.Default
```

Comment:
As an aside: @tnorling do you think it is time we identity B2C as its own `authorityType`?